### PR TITLE
feature/Extract learning rate scheduler as Class

### PR DESCRIPTION
**Pull request by yukkyo**
This is just an idea.

# Why
- easy to see slightly
- easy to use other learning rate scheduler

# Test
Not yet 😢 .

In my local, I changed `train.py` slightly and check `python train.py --weights_path weights/darknet53.conv.74` does not return error.

- add learning rate scheduler
```python
optimizer2 = optim.SGD(params, lr=base_lr, momentum=momentum,
                          dampening=0, weight_decay=decay * batch_size * subdivision)

lr_scheduler = MyLRScheduler(optimizer=optimizer2, base_lr=lr, burn_in=burn_in, decay_steps=steps,
                                 batch_size=batch_size, subdivision=subdivision)
```
- insert `lr_scheduler.step()` in each step
- compare two optimizer parameters by `utils.lr_scheduler.compare_optimizer_params()`

### CODE REVIEW

**Comment by hirotomusiker** [comment]
Thank you!
As for the scheduler, we would like to keep the train.py as it is.
In other repos we also write it using wrappers such as the chainer’s trainer, however the purpose of this repo is to make darknet understandable as possible - including lr settings. So could you possibly use the wrappers in your fork? 
And I recommend not to set lr at every iteration - it’s only necessary in 0.2% of the whole training steps.
Thank you again.

**Comment by yukkyo** [comment]
Thank you!
I understood about the background.
So I close this PR.
Thank you again.

Best regards
